runtime/tools/python/ttrt/common/run.py

```diff
@@ -407,10 +409,26 @@ def _execute(binaries):
                                     f"input_{i}"
                                 )
 
-                                if len(golden_tensor) != 0:
-                                    golden_inputs.append(
-                                        torch.tensor(golden_tensor, dtype=torch.float32)
+                                if golden_tensor is not None:
+                                    # Determine the element type of the golden tensor
+                                    match golden_tensor.dtype:
+                                        case DataType.Float32:
+                                            dtype = torch.float32
+                                        case DataType.UInt32:
+                                            dtype = torch.uint32
+                                        case DataType.UInt16:
+                                            dtype = torch.uint16
+                                        case DataType.UInt8:
+                                            dtype = torch.uint8
+                                        case _:
+                                            raise ValueError(
+                                                "Only F32 and unsigned integers are supported for `GoldenTensor`s"
+                                            )
```

Comment:
Maybe we should have a common utility function that can do this conversion?

Reply:
✅